browser/src/libs/code_intelligence/code_intelligence.tsx

```diff
@@ -830,78 +839,74 @@ export function handleCodeHost({
             console.log('Code view added')
             codeViewEvent.subscriptions.add(() => console.log('Code view removed'))
 
-            const { element, fileInfo, getPositionAdjuster, getToolbarMount, toolbarButtonProps } = codeViewEvent
-            const uri = toURIWithPath(fileInfo)
-            const languageId = getModeFromPath(fileInfo.filePath)
-            const model = { uri, languageId, text: fileInfo.content }
-            // Only add the model if it doesn't exist
-            // (there may be several code views on the page pointing to the same model)
-            if (!extensionsController.services.model.hasModel(uri)) {
-                extensionsController.services.model.addModel(model)
-            }
-            const editorData: CodeEditorData = {
-                type: 'CodeEditor' as const,
-                resource: uri,
-                selections: codeViewEvent.getSelections ? codeViewEvent.getSelections(codeViewEvent.element) : [],
-                isActive: true,
-            }
-            const editorId = extensionsController.services.viewer.addViewer(editorData)
-            const scope: CodeEditorWithPartialModel = {
-                ...editorData,
-                ...editorId,
-                model,
-            }
-            const rootURI = toRootURI(fileInfo)
-            addRootRef(rootURI, fileInfo.rev)
-            codeViewEvent.subscriptions.add(() => {
-                deleteRootRef(rootURI)
-                extensionsController.services.viewer.removeViewer(editorId)
-            })
+            const { element, diffOrFileInfo, getPositionAdjuster, getToolbarMount, toolbarButtonProps } = codeViewEvent
 
-            if (codeViewEvent.observeSelections) {
-                codeViewEvent.subscriptions.add(
-                    // This nested subscription is necessary, it is managed correctly through `codeViewEvent.subscriptions`
-                    // eslint-disable-next-line rxjs/no-nested-subscribe
-                    codeViewEvent.observeSelections(codeViewEvent.element).subscribe(selections => {
-                        extensionsController.services.viewer.setSelections(editorId, selections)
-                    })
-                )
-            }
+            const initializeModelAndViewerForFileInfo = (
+                fileInfo: FileInfoWithContent & FileInfoWithRepoNames,
+                useSelections?: boolean
```

Comment:
Boolean positional parameters are hard to make sense of at the call site. We often use the options pattern in similar cases:

```suggestion
                options?: { useSelections: true }
```

Thanks to optional chaining, this is any more inconvenient to use in the function body `if (options?.useSelections)`, but is much clearer at the callsite `initializeModelAndViewerForFileInfo(fileInfo, { useSelections: true })`.

Reply:
Sounds good. And this flag might change because I wasn't sure about the correct cases when selections should be enabled: only on a single file, or also on one (but not both?) viewers in a diff?

Reply:
You're right, this should change - we should just rely on the presence of `codeViewEvent.getSelections()` to determine whether to enable selections.